integration/update_test.go

```diff
@@ -460,3 +461,158 @@ func TestUpdateSet(t *testing.T) {
+		},
+
+		// TODO issues #673
+		/* "FieldDoc": {
+			filter: bson.D{{"_id", "1"}},
+			update: bson.D{{"$rename", bson.D{{"name", bson.D{{}}}}}},
+			err: &mongo.WriteError{
+				Code:    2,
+				Message: `The 'to' field for $rename must be a string: name: { : null }`,
+			},
+			altMessage: `The 'to' field for $rename must be a string: name: object`,
+		}, */
+
+		"FieldArray": {
+			filter: bson.D{{"_id", "1"}},
+			update: bson.D{{"$rename", bson.D{{"name", primitive.A{}}}}},
+			err: &mongo.WriteError{
+				Code:    2,
+				Message: `The 'to' field for $rename must be a string: name: []`,
+			},
+			altMessage: `The 'to' field for $rename must be a string: name: array`,
+		},
+		"FieldNaN": {
+			filter: bson.D{{"_id", "1"}},
+			update: bson.D{{"$rename", bson.D{{"name", math.NaN()}}}},
+			err: &mongo.WriteError{
+				Code:    2,
+				Message: `The 'to' field for $rename must be a string: name: nan.0`,
+			},
+			altMessage: `The 'to' field for $rename must be a string: name: double`,
+		},
+		"FieldNil": {
+			filter: bson.D{{"_id", "1"}},
+			update: bson.D{{"$rename", bson.D{{"name", nil}}}},
+			err: &mongo.WriteError{
+				Code:    2,
+				Message: `The 'to' field for $rename must be a string: name: null`,
+			},
+			altMessage: `The 'to' field for $rename must be a string: name: null`,
+		},
+		"RenameString": {
+			filter: bson.D{{"_id", "1"}},
+			update: bson.D{{"$rename", "string"}},
+			err: &mongo.WriteError{
+				Code: 9,
+				Message: `Modifiers operate on fields but we found type string instead.` +
+					` For example: {$mod: {<field>: ...}} not {$rename: "string"}`,
+			},
+			altMessage: "Modifiers operate on fields but we found type string instead. " +
+				"For example: {$mod: {<field>: ...}} not {$rename: string}",
+		},
+		"RenameNil": {
+			filter: bson.D{{"_id", "1"}},
+			update: bson.D{{"$rename", nil}},
+			err: &mongo.WriteError{
+				Code: 9,
+				Message: `Modifiers operate on fields but we found type null instead.` +
+					` For example: {$mod: {<field>: ...}} not {$rename: null}`,
+			},
+			altMessage: "Modifiers operate on fields but we found type null instead. " +
+				"For example: {$mod: {<field>: ...}} not {$rename: null}",
+		},
+		"RenameDoc": {
+			filter: bson.D{{"_id", "1"}},
+			update: bson.D{{"$rename", primitive.D{}}},
+		},
+
+		// TODO issues #673
+		/* "RenameDoc_1": {
+			filter: bson.D{{"_id", "1"}},
+			update: bson.D{{"$rename", bson.D{{}}}},
+			err: &mongo.WriteError{
+				Code:    56,
+				Message: `An empty update path is not valid.`,
+			},
+			altMessage: `An empty update path is not valid.`,
+		}, */
+	} {
+		name, tc := name, tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			ctx, collection := setup(t)
+
+			_, err := collection.InsertMany(ctx, []any{
+				bson.D{{"_id", "1"}, {"name", "alex"}, {"phone", "9012345678"}},
+				bson.D{{"_id", "2"}, {"name", "bob"}},
+			})
+			require.NoError(t, err)
```

Comment:
I think that it becomes a problem that, unlike `find` tests, each `update` test uses its own data set. Let's put that PR on pause until we decide if we want to do something about that. Created #674 for tracking

Reply:
Let's hook into update_field_compat_test.go instead of using test-specific data set